### ISSUE TITLE
Course, CourseInstance, EducationEvent [Schema.org Course Extension]

### ISSUE THREAD

**Issue by westurner** [schemaorg/schemaorg#195]
- [x] Thing > CreativeWork > Course https://schema.org/Course
- [x] Thing > Event > CourseInstance https://schema.org/CourseInstance
- [x] Thing > Event > EducationEvent https://schema.org/EducationEvent

.

- | Project: https://www.w3.org/community/schema-course-extend/
- | MailingList: https://lists.w3.org/Archives/Public/public-schema-course-extend/
- | Wiki: https://www.w3.org/community/schema-course-extend/wiki/
- | Issue: https://github.com/schemaorg/schemaorg/issues/195
- | PR: https://github.com/schemaorg/schemaorg/pull/972
- | RDFa schema & examples: https://github.com/schemaorg/schemaorg/blob/sdo-deimos/data/ext/pending/issue-195.rdfa
- | Web: http://course.schema-course-extend.appspot.com/
- | Web: http://pending.webschemas.org/Course
- | Web: http://pending.webschemas.org/CourseInstance
- | Web: https://schema.org/Course
- | Web: https://schema.org/CourseInstance

Goals:
- schema.org/Course , schema.org/CourseSection , schema.org/EducationEvent 

> SCHEMA COURSE EXTENSION COMMUNITY GROUP
> 
> This mission of this group, initiated by LRMI, is to develop an extension for schema.org concerning the discovery of any type of educational course (online/offline, long/short, scheduled/on-demand). Educational course is defined as "some sequence of events and/or creative works which aims to build the knowledge, competence or ability of learners". (Out of scope: information about students and their progression etc; information needed internally for course management rather than discovery).
> 
> -- https://www.w3.org/community/schema-course-extend/

Products:
- [x] DOC: "Schema.org: Online Courses" [Schema]  (@vholland, et al.)
  https://goo.gl/2Sxxy3
  - Thing > Intangible > Course
  - Thing > Intangible > CourseSection
  - Thing > Event > EducationEvent
  - [UPDATE]
  - [{ }] Thing > Intangible > Course
  - [{ }] Thing > CreativeWork > Course
  - [{ }] Thing > Product > Course
- [x] DOC: "Schema for Courses" [Use Cases] (@philbarker, et al.)
  https://goo.gl/DnTTsb
  - [x] https://www.w3.org/community/schema-course-extend/wiki/Outline_use_cases
- [x] ENH: `data/ext/course/course.rdfa` -- Course extension RDFa
- [ ] ENH: `data/ext/course/course-course-examples.txt` -- course extension examples
- [ ] TST: equiv of `make test` and `make build`
- [ ] DOC: release notes
- [x] DOC: blog post: http://blog.schema.org/

Examples:
- http://www.w3.org/wiki/WebSchemas/Course
- https://sites.google.com/site/moocontology/home
- https://sites.google.com/site/moocontology/onlinecourse
- https://sites.google.com/site/moocontology/example

Discussion Mailing List Threads:
- **http://lists.w3.org/Archives/Public/public-vocabs/2014Oct/0052.html**
  
  "[Proposal] schema:OnlineCourse"
- http://lists.w3.org/Archives/Public/public-vocabs/2014Feb/0127.html
  - http://lists.w3.org/Archives/Public/public-vocabs/2014Feb/0131.html
- http://lists.w3.org/Archives/Public/public-vocabs/2012Dec/0032.html
- http://lists.w3.org/Archives/Public/public-vocabs/2012Nov/0004.html

Use Cases / Potential Implementors (Culled from the Mailing List):
- https://www.khanacademy.org/
- http://ocw.mit.edu/
- http://www.ieee.org/portal/innovate/products/educational/
- http://www.acm.org/education/curricula/ComputerScience2008.pdf
- https://www.udacity.com/
- https://www.edx.org/
- https://www.coursera.org/
  - _https://www.coursera.org/course/datascitoolbox_ \* Proposal Example
- http://www.edumine.com/
- https://www.futurelearn.com/
- https://en.wikipedia.org/wiki/List_of_learning_management_systems
- [University] Course Catalogs contain prerequisites and matriculation requirements
  - https://explorecourses.stanford.edu
  - http://student.mit.edu/catalog/m6a.html
    - _http://www.mit.edu/catalog/6.034_ \* Proposal Example
  - http://www.ccae.org/
    - _http://learn.ccae.org/modules/shop/index.html?action=section&OfferingID=62&SectionID=1486_ \* Proposal Example
  - http://data.wu.ac.at/
  - http://data.wu.ac.at/dataset/all_courses_2014s
  - http://data.wu.ac.at/dataset/all_course_events_2014s
- US Primary education standards: http://www.corestandards.org/ "Common Core"
  - This could potentially be extremely useful for discovering useful curriculum resources
- https://developers.google.com/edu/curriculumsearch/ (educationalAlignment -> "Course")
- https://www.class-central.com/
  - https://github.com/dhawalhshah/class-central/blob/master/src/ClassCentral/SiteBundle/Resources/views/Default/courses.table.html.twig
  - https://github.com/dhawalhshah/class-central/blob/master/src/ClassCentral/SiteBundle/Resources/views/Course/mooc.html.twig
- https://github.com/edx/
  - https://github.com/edx/edx-platform/blob/master/cms/djangoapps/models/settings/course_details.py
  - https://github.com/edx/edx-platform/blob/master/common/djangoapps/course_modes/models.py
  - https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/course_about.html
  - https://github.com/edx/edx-platform/blob/master/lms/templates/courseware/courses.html
    - https://github.com/edx/edx-platform/blob/master/lms/templates/course.html

Release notes:
- http://blog.schema.org/2017/03/schemaorg-32-release-courses-fact.html
- https://schema.org/docs/releases.html#v3.2

**Comment by jgoodell2** [comment]
Building on Vicki's work I suggested changes to the Google doc (1) based on what seems to be areas of consensus on the discussion threads and comments within the doc. 

I'm afraid that there are different threads/communities with definitions that may work fine within a narrow purpose, but have conceptual errors when trying to apply the vocabulary more universally. The approach discussed in the group seems to be headed in the right direction. How can we bring the different communities together on this? 

For example, moocontology (2) has "instructor" and "event" as properties of Course rather than defining a CourseSection. This can create ambiguity when multiple Course Sections are offered at the same or different times. e.g. This month EdX started a Section of "Introduction to Computational Thinking and Data Science" (course number 6.00.2x). This is not the first time the course was offered. The course content/description is the same for each instance, but the start and end dates change, and the instructors may change over time. There could be Sections going on concurrently with different instructors, but the same course number, learning objectives, description. 

I know there are discussions about Course as an extension to schema.org, rather than core; even so, I  think we should try to get the different communities together so Course isn't defined as different things by different communities.

-jim

(1) https://docs.google.com/document/d/12YWjLzZC8FiTiOwSAETRIEozeqZdn6O8a4fgqK4t5Ss/edit?usp=sharing
(2) https://sites.google.com/site/moocontology/

**Comment by westurner** [comment]
Is it CourseSession or CourseSection?

**Comment by westurner** [comment]
Is it CourseSession or CourseSection?

(It was CourseSession, but was changed without discussion to CourseSection).

* I vote for "CourseSession". To me, Section implies a physical partitioning; which does not fit.

**Comment by parklize** [comment]
+1 to "CourseSession", Coursera and Stanford MOOC use "session".

Coursera API: https://tech.coursera.org/app-platform/catalog/
Stanford Online: http://online.stanford.edu/courses/all

On Fri, Mar 6, 2015 at 12:00 AM, Wes Turner <notifications@github.com>
wrote:

> Is it CourseSession or CourseSection?
>
> (It was CourseSession, but was changed without discussion to
> CourseSection).
>
>    - I vote for "CourseSession". To me, Section implies a physical
>    partitioning; which does not fit.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/schemaorg/schemaorg/issues/195#issuecomment-77479627>.
>



-- 
-----------------------------------------------------------------------------------
To be Sharer^^

A man is not old as long as he is seeking something.
A man is not old until regrets take the place of dreams.

Email: parklize@gmail.com
----------------------------------------------------------------------------------

**Comment by stuartasutton** [comment]
CourseSession and CourseSection are totally different beasts. A Course may have many CourseSections and CourseSections may have many CourseSessions. For example, at the Univ. of Washington (Seattle) there is a course in the University inventory (catalog) called "Metadata Design Studio". This spring quarter, there are two CourseSections of this Course (SLN 15343 and SLN 15344). This week, (the section I teach) SLN 15344 is in the second CourseSession of the spring quarter. I am less concerned about what Coursera and the Stanfod MOOC call things than with the pattern that holds in the nomenclature of thousands of colleges and universities--at least in the U.S.

**Comment by philbarker** [comment]
hello. I have been thinking about the scope and use cases for this proposal. The discussion above starts with a bullet point referring to "online courses" but I think we have gone beyond that. Also I can see lots of potential implementers above, but nothing about use cases that relate to what a user might want to do.
My initial ideas are at http://goo.gl/DnTTsb I would welcome comments/corrections. Apologies if they are sketchy, I wanted to share early rather than make unwarranted assumptions.

**Comment by philbarker** [comment]
hello. I have fleshed out the ideas I mentioned before for more specific scope and use cases, still in google doc at http://goo.gl/DnTTsb . Any comments? I think it would be useful to compare the proposals to these use cases and some examples, but it'ls also be good to have some agreement that the use cases are valid and nothing major is missing before doing that.

Is a google doc the best place to do this, would you rather have it on github somewhere?

**Comment by westurner** [comment]
@philbarker I've updated the ticket description w/ the link to your google doc containing use cases for these vocabulary extensions.

I think it would be great to build the tests for these use cases (in JSON-LD, RDFa, or just RDF TTL/N3). (the JSON-LD test suite uses http://json-ld.org/test-suite/reports/ uses ``earl`` for similar reports: http://www.w3.org/TR/EARL10-Schema/ )

**Comment by philbarker** [comment]
Thanks @westurner . I do intend to cross check the use cases against some of the example courses listed above and in the scoping part of the google doc and against the current proposal. I'm not sure that my technical skills extend to building tests along the lines you point to, or what data there is to test (none yet, or maybe I am misunderstanding?)

**Comment by westurner** [comment]
Thanks! Really just JSON-LD and RDFa examples that demonstrably (and
recordedly) evaluate to valid RDFa.

To be clear, I opened this ticket but by no means should this be construed
to indicate leadership on this issue. (It may be helpful to start reaching
out for more input from e.g. a registrar's office: I believe we're
currently missing a way to represent prerequisite edges)

**Comment by danbri** [comment]
@westurner - Phil and I had a quick chat about prerequisites last week. He pointed out that we should try to have a mechanism for this which is compatible with our approach to representing learning outcomes. Doing so will make it easier to chain together learning opportunities from a larger pool...

**Comment by danbri** [comment]
BTW please note that we now have a new W3C Community Group devoted to Course description: https://www.w3.org/community/schema-course-extend/ 

From schema.org's point of view this issue #195 is probably the best interface between schema.org in general and Course-specific issues, so let's try to keep this issue up to date with any major progress or discussion points, even if more activity moves into the new and more specific Community Group.

**Comment by westurner** [comment]
> From schema.org's point of view this issue #195 is probably the best interface between schema.org in general and Course-specific issues, so let's try to keep this issue up to date with any major progress or discussion points, even if more activity moves into the new and more specific Community Group

* | Project: https://www.w3.org/community/schema-course-extend/
* | MailingList: https://lists.w3.org/Archives/Public/public-schema-course-extend/ "by thread"

**Comment by danbri** [comment]
I've posted a *snapshot* of the current community group design in the pending extension,  http://webschemas.org/docs/releases.html#g195 ... at this stage without examples. The source was https://github.com/philbarker/schemaorg-1/tree/feature/ext-course/data/ext/course

For more about "pending" (a new development) see http://webschemas.org/docs/howwework.html#pending

**Comment by danbri** [comment]
I've updated the version in pending so that grantsCredential does not reference a non-existent "Credential" type (and we have updated our unit tests to run against the extension schemas so bugs like that won't slip past in future). However I would encourage use of other terms here, as talk of granting credentials may sound like a crypto protocol to non-educationalists.

**Comment by westurner** [comment]
Why not add
Intangible > Credential
CreativeWork > Credential
Credential > Certificate > Diploma
?

There is a need for a base Credential type in schemed, as well:
 https://github.com/twamarc/ScheMed/issues/34#issuecomment-213594529

At a later time, it may make perfectly good sense to add as subclasses of
Credential:

* ...
* SSLCertificate
* GPGKey
* SSHKey
* WebID
* ...

**Comment by westurner** [comment]
To focus more on modeling Offers for Services and a Product, IDK how to do:

* Credential > Degree
* Credential > Diploma

**Comment by danbri** [comment]
I don't see SSLCertificate, GPGKey etc  and Degree, Diploma having a lot in common. I have a degree from the University of Bristol, and numerous SSLCertificates from various sources, but they do not have much in common. Let's please not mix anything crypto-related casually into other topics. I don't often express strong opinions on schema design here but on this issue I'm pretty sure we would quickly come to regret such an entangling of issues.

**Comment by Dataliberate** [comment]
+1

However the raising of this highlights the need to choose term names with care to avoid future confusion and wrong assumptions. 

~Richard

**Comment by stuartasutton** [comment]
Yes, please, I am with Dan in regards to not doing this kind of
mixing/polluting. There is a huge, loosely related set of global verticals
made up of for-profit and non-profit credentialing organizations with
products as well as millions of people who have earned credentials publicly
denoting success in some endeavor.

There is work well underway on development RDF-based, rich description of
credentials of these more formal sorts [1]-[2] as well as W3C work on
credentials and verifiable claims regarding them [3]-[4]. Let's do keep
these more formal (and widely) used notion of credential distinct from
"anything crypto-related". There are hopes within CTI that a light-weight
version of the work at [1]-[2] will form the basis for a Credential type
within schema.org.

Stuart

[1] http://www.credentialtransparencyinitiative.org/
[2] http://www.credreg.net/
[3] https://www.w3.org/community/credentials/
[4]
https://www.w3.org/Payments/IG/wiki/Main_Page/ProposalsQ42015/VerifiableClaimsTaskForce

**Comment by westurner** [comment]
@danbri 
> I don't see SSLCertificate, GPGKey etc and Degree, Diploma having a lot in common. 

These are all types of credentials.
* A Key is a Credential.
* A Certificate is a Credential.
* A Diploma is a certificate (which references a Degree)
* A Degree is a Credential.

At this time, we have no Creative-Commons schema.org way to define Credential > {Certificate, Degree}.

Use Cases ( @philbarker ) :

* (Online) Course providers can list Credentials {Certificates, } which can be earned by satisfactorily passing an Offered Course.
* Educational Organizations can list Degrees which they Offer.

...

(Out of scope, but still logically deriving from a base Credential class):

* A Credential may be cryptographically signed (e.g. by a PKI)
  * In which case, there must be:
    * a field for the hash
    * a field for a fingerprint
    * a field for the signingAuthority Person/Organization
    * schema.org/Person / schema.org/Organization URL (schema.org/url // rdf:subject)
      and/or denormalized metadata for signing
  * OpenBadges works today.
    * https://github.com/openbadges/openbadges-specification/issues/9
      * [x] JWS in JSONLD
      * [ ] JWS in RDFa 

I believe a preliminarily defined Credential > Degree *is* in scope for this effort in order to define a ``rdfs:range`` of ``grantsCredential``.

Blocking objectives for Schema.org Course Extension
* It doesn't make sense to release Course until we can also describe the Credential which can be earned from a given Course.
* It doesn't make sense to release Course until we can also describe **physicalAccessibilityFeature** for Courses (as well as Events, Places, and -- arguably -- CreativeWorks)
* (Implementers aren't going to want to go back and re-work this into the public-facing HTML RDFa templates for Courses (and Degrees) which can be generated from a student/course information system)

...

@stuartasutton 

> Credential
> [...]
> Questions for researching potential/ideal implementations of schema.org/Credential and e.g. schema.org/Degree:
> 
> * Are you open to **Creative Commons** as a license [hard schema.org requirement]?
> * Are you open / friendly to other application implementations?
> * Is the/your application open source? (link to source: ________; early and often)
> * Do you [already] solve for **DegreeProgram**?

(DegreeProgram will/would require conditionals)

**Comment by stuartasutton** [comment]
On Sat, May 21, 2016 at 5:54 AM, Wes Turner <notifications@github.com>
wrote:

>
> I believe a preliminarily defined Credential > Degree *is* in scope for
> this effort in order to define a rdfs:range of grantsCredential.
>
I believe the rdfs:range (schema:rangeIncludes) of grantsCredential should
remain Credential. Degree is one subclass of Credential that could be
applicable in the context of an instance of Course; but, not the only one.
See
https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl.
A number of these subclasses could be applicable to Course.

> @stuartasutton <https://github.com/stuartasutton>
>
If you mean by "you" the CTI initiative (which is not me), the following
responses hold:

> Credential
> [...]
> Questions for researching potential/ideal implementations of
> schema.org/Credential and e.g. schema.org/Degree:
>
>    - Are you open to *Creative Commons* as a license [hard schema.org
>    requirement]?
>
> Creative Commons licensing is not a "hard requirement". The CTI
specifications under development will all have open licenses (e.g.,
https://creativecommons.org/licenses/by/4.0/).

>
>    - Are you open / friendly to other application implementations?
>
> Absolutely. There are a number of 3rd party implementers advising the CTI
work because they know that that work is open from the start and is
intended to spur a marketplace of applications. While CTI, under its grant
funding, is developing a Credential Registry (also to be openly licensed)
and a set of applications for interacting with it (also open), there are no
illusions that it will be the only credential registry or the only set of
applications developed that will act on data based on the Credential
Transparency Description Language (CTDL). So, it is very open to the
developer community as part of its strategy for long-term sustainability.

>
>    - Is the/your application open source? (link to source: ________;
>    early and often)
>
> Not yet. But it will be (on GitHub) when there is something to put in
front of people. The applications and the Credential Registry are in early
stages of development. It's too early to know whether CTI will maintain an
actual open source community (not easy to do); but, it is not too early to
know that the code base for the apps CTI is developing under its grant
funding will be openly available through GitHub.

>
>    - Do you [already] solve for *DegreeProgram*?
>
> (DegreeProgram will/would require conditionals)
>
If you mean that an instance of Credential (or one of its subclasses) stems
from a degree program (obviously, not all do), then the conditions for
granting the degree are within scope.

**Comment by westurner** [comment]
* Can we define Credential and Degree today? Or, are we blocked pending a partial third-party implementation?
* IIUC, CC-By-SA [3.0,] is a hard (viral) requirement for schema.org and contributions
  From https://schema.org/docs/terms.html :

  > **Scope of Terms of Services; License**
  > These Terms of Service govern your use of the Website, which contains a schema specifying a vocabulary you can use. The Sponsors' copyrights in the schema are licensed to website publishers and other third parties under the Creative Commons Attribution-ShareAlike License (version 3.0). To view a copy of this license, please visit http://creativecommons.org/licenses/by-sa/3.0/. The Sponsors have applied the W3C Patent Policy to the schemas published by Schema.org as follows: each Sponsor, by itself and on behalf of its affiliates, agrees to make available under W3C RF licensing requirements its Essential Claims (if any) in the schemas published by Schema.org as if the schemas were W3C Recommendations. In some cases, this website may indicate that some but not all of the Sponsors have recognized a particular extension to the Schema; in those cases, as to that extension, the above rights are granted by only those recognizing Sponsors.

* From https://wiki.creativecommons.org/wiki/4.0_upgrade_guidelines :

  > **Existing content:**
  > - Who owns the rights?
  >   - If the publisher, then can relicense under 4.0 as specified above.
  >     - If the contributors, then need permission to relicense. Without permission (via terms of use or otherwise), then that content remains under prior version. [If this is the case, see the section below about dealing with mixed-version content.]
  >       - A couple of options for obtaining permission to relicense:
  >       - upon upload by contributors, have a prompt box to obtain agreement to relicense previous uploads;
  >       - general outreach to contributors seeking agreement to upgrade. [Note that this is easier to do with discrete artifacts (an article, a photo) as opposed to other contributions such as comments on wikis and similar, where one person's contribution is intermixed with others.]

**Comment by westurner** [comment]
> I believe the rdfs:range (schema:rangeIncludes) of grantsCredential should
remain Credential. Degree is one subclass of Credential that could be
applicable in the context of an instance of Course; but, not the only one.

@danbri 
- Is this the file I should be appending to now? (IDK how to maintain the revision log when ``git mv``'ing files around in a git repository):
 https://github.com/schemaorg/schemaorg/blob/sdo-deimos/data/ext/pending/issue-195.rdfa#L120
 (was: https://github.com/schemaorg/schemaorg/pull/972/commits )
  - I ask because from here https://github.com/schemaorg/schemaorg/issues/195#issuecomment-220588244 

    > I've updated the version in pending so that grantsCredential does not reference a non-existent "Credential" type 

   Where is this change?

- I think we still need to do the ``<!-- TODO: define Credential > [Certificate, ] -->`` part
  except actually it's now {Intangible || CreativeWork?} > Credential > {Certificate, Degree}

**Comment by westurner** [comment]
> (IDK how to maintain the revision log when git mv'ing files around in a git repository)

If you include either a complete URL to a revision when copy-pasting OR just a shortrev, GitHub will include the link in the commit message (in order to maintain traceability between PR branches, commits, etc)

Two ways to print the shortrev (to keep commit messages usable and minimal):

```bash
git log --abbrev-commit
git rev-parse --short HEAD
```

- URL: ``https://github.com/schemaorg/schemaorg/commit/62086c8d6658e7e7ebed8993930ad1ba8ea2b4df``
- URL: https://github.com/schemaorg/schemaorg/commit/62086c8d6658e7e7ebed8993930ad1ba8ea2b4df
- shortrev: ``62086c8`` (first 7 characters)  

So GitHub would linkify a commit message like:

```
Added a first snapshot of the ongoing courses work. (from 62086c8 )
```

**Comment by philbarker** [comment]
Personally I am quite happy with the term name Credential. It doesn't immediately suggest anything crypto to me (I think the wikipedia article gets the coverage about right[1])

I also think that any meaningful alternative I can think of has similar problems. The best I can think of is grantsAward -- is anyone happier with that?

I also don't think that the course extension work is a good place to try to pin down a controlled vocabulary of all the possible Awards.


1. https://en.wikipedia.org/wiki/Credential

**Comment by westurner** [comment]
> 1. https://en.wikipedia.org/wiki/Credential

Minimally, should this initial Credential RDFS class be derivative of **Intangible** or of **CreativeWork**?:

* OR
  * Thing > Intangible > Credential
  * Thing > CreativeWork > Credential

**Are there any base properties which are specific to Credential?**
I would think that there would then, eventually, from a different schema.org extension group, be a CryptographicCredential (where x.509 attributes may cover for most other formats: https://en.wikipedia.org/wiki/X.509#Structure_of_a_certificate (though, again, this is not yet in scope for academic credentialing, pending the recommendations of the CTI and credreg))

Brainstorming eventual Credential types:

* Credential
  - Certificate
  - Certification
  - Degree
  - Diploma (Degree)
  - MedicalSpecialty / MedicalCredential (https://github.com/twamarc/ScheMed/issues/34)

**Comment by ottonomy** [comment]
Excited to see some work on this. Came over from https://github.com/openbadges/openbadges-specification/issues/9 as I was doing a little pruning. 

Would be worth paying attention to the VCTF discussion of denoting their class proposal as "WebCredential" or "DigitalCredential" to make it more specific and differentiate from more general approaches to "Credential", like schema.org is likely to take.

I'd probably lean toward grouping Credential under CreativeWork. 

@stuartasutton may know more on proposed types for credentials. The CTI meeting from a week ago previewed the first snippet of a list of proposed types, but I wasn't able to find the list on GitHub as was hinted.

**Comment by stuartasutton** [comment]
@ottonomy, my sense of this is that schema.org/Credential is correct since it aligns with the CTI definition as the CreativeWork that describes the thing created by a credentialing entity (of any sort) with the intention that it be awarded. The VCTF context uses the (class name still under consideration) "TBD Credential" as a verifiable set of claims. A somewhat subtle but very significant difference.

CTI has a (unstable) set of subclasses for Credential as well as an (equally unstable) enumeration of credential types of concern to CTI at https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl

+1 for Credential as CreativeWork.

**Comment by westurner** [comment]
> https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl

Thanks! Looking forward to learning more here.
It would be great if we could get this together for Universities to catalog and implement
(probably with spreadsheets)
by this fall.

### rdf:type ('a', 'is a', 'hasType') || rdf:Property ('hasProperty')

* OR (hasProperty credType || hasType CredType) ?
  * With schema org RDFS, IIUC, these credType:credential skos:concepts (linked with skos:broader and skos:narrower) would be TitleCased rdfs:Class subclasses of e.g schema:CreativeWork
    * Thing
      * CreativeWork
        * Credential
          * Degree
            * AssociateDegree
            * BachelorDegree
            * GraduateDegree
            * DoctoralDegree

  * Credential.credType is an rdfs:Property (attribute, predicate) with an rdfs:range of Credential and an rdfs:domain of credType (r: Credential, d: credType)

    * Modeling with "hasProperty [IRI]" is less flexible than "hasType [IRI]" because hasProperty assumes that all Credential types have the same attributes.
      * Here that may not be true?
        * e.g. .thesis/.dissertation may be a property of certain degrees
          (d:{GraduateDegree, DoctoralDegree} r:CreativeWork)
    * Modeling with hasProperty is more flexible than hasType because
      sometimes it doesn't make sense for things to be subclasses of
      multiple intersecting types.

      * Thing > CreativeWork > Credential > Degree > BachelorDegree
      * Thing > CreativeWork > Credential > Degree > BachelorDegreeInstance


### Concept / Instance differentiation
* How to differentiate between an offered Credential and an earned instance of a Credential?
  It would seem that here, as well, we'll need such a distinction?

  * A University Offers Degrees and focusAreas
  * An Indivdual has a Degree with focusAreas

    * DegreeInstance? (REQ: terminology)
      - credType [Degree **Enumeration** URI]


###  Degree-specific terminology

- schema:about d: https://schema.org/CreativeWork r: https://schema.org/Thing
- focusArea
- academicDiscipline
- 
- REQ: proper terminology
- NOTE: Wikipedia URLs (and dbpedia URIs) make good (human-readable) concept URIs; but are not necessarily *stable* (because disambiguation)


#### an example

Thing > CreativeWork > Degree > BachelorDegree
- credType: bachelorDegree / BachelorDegree \*
- ?: Bachelor of Science (BSc, B.Sc., ...)
- ?: Management Information Systems (MIS)
- focusArea: Quantitative Analysis
- focusArea: Entrepreneurship

.

* https://en.wikipedia.org/wiki/Bachelor%27s_degree#Computer_science_and_information_systems

**Comment by westurner** [comment]
Does there need to be another layer here between
[...] > Credential > Degree
?

**Comment by westurner** [comment]
Does there need to be another layer here between
Credential [... > ...] Degree
?

Because:
- Are there other systems of academic credentials?
- WebCredential / DigitalCredential ?

**Comment by danbri** [comment]
Aside, http://blockchain.open.ac.uk/ is worth keeping an eye on.

**Comment by westurner** [comment]
> Aside, http://blockchain.open.ac.uk/ is worth keeping an eye on.

- https://Keybase.io/ stores signatures in at least one blockchain; as well as centrally.
  - https://keybase.io/docs/server_security/merkle_root_in_bitcoin_blockchain

From https://en.wikipedia.org/wiki/Merkle_tree ([oldid=720708959](https://en.wikipedia.org/w/index.php?title=Merkle_tree&oldid=720708959#Uses)) :

> Suggestions have been made to use hash trees in trusted computing systems.[4] Hash trees are used in the IPFS and ZFS file systems,[5] BitTorrent protocol, Apache Wave protocol,[6] Git distributed revision control system, the Tahoe-LAFS backup system, the Bitcoin peer-to-peer network, the Ethereum peer-to-peer network,[7] the Certificate Transparency framework, and a number of NoSQL systems like Apache Cassandra and Riak

**Comment by philbarker** [comment]
@westurner "Are there other systems of academic credentials?"

Yes, there are.See [https://en.wikipedia.org/wiki/National_Qualifications_Framework#England.2C_Wales_and_Northern_Ireland](https://en.wikipedia.org/wiki/National_Qualifications_Framework#England.2C_Wales_and_Northern_Ireland) for a list of Baccs, Certs, BTECs, Diplomas, Awards, PGCerts etc. in England, Wales and Northern Ireland (Scotland is different)

There is also the whole world of industry awards, e.g. CISCO/Microsoft Certifications but not limited to IT. 

Phil

**Comment by danbri** [comment]
yes, it would be unfortunate if we could only express academic but not
vocational qualifications. Also driving licences (for trucking etc.),
acting and other fields this can be seen in terms of organisational
membership (eg http://www.equity.org.uk/about-us/join-us/how-can-i-join/
http://fia-actors.com/no_cache/about-fia/fia-members/ ). see also
https://pending.schema.org/WorkersUnion but also other professional
membership structures.

**Comment by danbri** [comment]
re credentials, take a look at https://www.google.com/trends/explore#q=credentials

related searches (at google):

windows credentials 	
user credentials 	
login credentials 	
network credentials 	
invalid credentials 	
credential 	
what is credentials

**Comment by westurner** [comment]
On Thursday, June 2, 2016, Dan Brickley <notifications@github.com> wrote:

> re credentials, take a look at
> https://www.google.com/trends/explore#q=credentials
>
There are different results for the @en singular form ("credentials" /
"credential")
https://www.google.com/trends/explore#q=credential

Minimally,
C: CreativeWork > Credential
- P: issuer R: Person/Organization

C: CreativeWork > Credential > CredentialInstance
- P: issuer R: Person/Organization
- P: issuee R: Person/Organization

Or,
P: CredentialInstance.credential R: Credential

Are there more appropriate terms than issuer/issuee?

>

**Comment by westurner** [comment]
And/Or,
P: issuerCredential R: Credential D: {Credential, CredentialInstance}

**Comment by parklize** [comment]
How about issue as a more specific ["CreateAction" ](https://schema.org/CreateAction)?

**Comment by stuartasutton** [comment]
@danbri and @philbarker,  Is this discussion of defining a schema.org Credential  best placed here under an issue focused on Course? There are many who have vested interests in how schema.org might ultimately handle a definition of Credential who aren't necessarily watching Course issues.

**Comment by westurner** [comment]
On Friday, June 3, 2016, Stuart Sutton <notifications@github.com> wrote:

> @danbri <https://github.com/danbri> and @philbarker
> <https://github.com/philbarker>, Is this discussion of defining a
> schema.org Credential best placed here under an issue focused on Course?
> There are many who have vested interests in how schema.org might
> ultimately handle a definition of Credential who aren't necessarily
> watching Course issues.
>
Degree, Certificate, and Diploma are all blocked on a Credential base class.

Thing > CreativeWork > Credential > Degree
Thing > CreativeWork > Credential > Diploma
Thing > CreativeWork > Credential > CredentialInstance > Certificate >
Diploma
Thing > CreativeWork > Credential > Certification


There are many potential implementations of DigitalCredential /
CryptoCredential / _Credential.
When those rdfs:classes are defined in terms of schema.org RDFS, it should
be possible to describe types of Credentials and CredentialInstances
 through subclassing and/or composition.

Can we agree to work with 'issuer' and 'issuee' with domains of
{Credential, CredentialInstance}?

Degree [> BachelorDegree]
- [credType: cti:bachelorDegree]
- ?: Bachelor of Science
- name: Management Information Systems
- issuer: Organization(name="University of Nebraska Omaha") #orgX

So that a record can be signed, eventually there must be a way to do:

Diploma (or "DegreeInstance", or {Degree, CredentialInstance})
- issuerCredential: orgX.credential[n]
- issuerCredential: CredentialInstance(issuee=orgX)
- [generic signed signature term]

Again, OpenBadges JWS works today and KeyBase.io proofs work today.

**Comment by westurner** [comment]
On Friday, June 3, 2016, Guangyuan Piao <notifications@github.com> wrote:

> How about issue as a more specific "CreateAction"
> <https://schema.org/CreateAction>?
>

Maybe ConferAction?

Action > CreateAction > ConferAction

ConferAction
- agent: Organization orgX (CredentialInstance orgX.credential[n]
- result: CredentialInstance credX
- participant: Person personX
- startTime: DateTime iso8601datetimeX

DegreeInstance (or {Degree, CredentialInstance}) credX
- dateCreated: DateTime iso8601datetimeX
- issuee: Person personX

**Comment by stuartasutton** [comment]
The CTI project is uses subclasses of schema.org/Action with key actions
between agents (People/Organizations) and credentials and other
agents--chiefly in the context of quality assurance; e.g.,

AccreditAction
ApproveAction
ConferAction
EndorseAction
RecognizeAction
RenewAction
RevokeAction

CTI is deliberately focused on description of Credential as a work
(CreativeWork) and not on the credential awarded (or on related
transactions on such awarded artifacts (verifiable claims etc.)).

**Comment by CTITechPlanning** [comment]
Hi everyone. This is Jeanne with the Credential Transparency Initiative (CTI).  We've been following the progress with Course schema. In regards to the recent comments here on "Credential", the CTI defines this as, "A qualification or achievement of an agent."  As an additional comment, for CTI, only credentials issued to people, and credentials for quality issued to organizations for related programs, courses, tasks and/or assessments are in scope. As Stuart noted,  CTI metadata modeling takes into account the work of Schema.org.  The editor's draft classes, properties, and enumerations and other information is available from the CTI tech planning site http://credreg.net.

**Comment by westurner** [comment]
> CTI is deliberately focused on description of Credential as a work
(CreativeWork) and not on the credential awarded (or on related
transactions on such awarded artifacts (verifiable claims etc.)).

* “What we learned from designing an academic certificates system on the blockchain” @medialab https://medium.com/mit-media-lab/what-we-learned-from-designing-an-academic-certificates-system-on-the-blockchain-34ba5874f196
* “Verifiable Credentials on the Blockchain” @chrisjagers https://medium.com/learning-machine-blog/blockchain-credentials-b4cf5d02bbb7
* http://certificates.media.mit.edu
  * https://github.com/digital-certificates/cert-schema "**Cert-schema**: Cert-schema details how to make a digital certificate. A digital certificate is essentially a JSON file with the necessary fields needed for our cert-issuer code to place it on the blockchain. We tried to keep the schema as close to the Mozilla Open Badges specifications as possible."


... In terms of set logic (preferably as one or more Venn diagrams, if possible and you have the time), how can we support multiple Credentials?
- There may be (namespace-ignorant) Class name collision between e.g. schema:Credential, schema:CredentialInstance (?) and more complete, [blockchain-]verifiable schema like OpenBadges, cert-schema, and CTI

**Comment by westurner** [comment]
I've defined Credential, CredentialInstance, CredentialType, etc here: https://github.com/westurner/schemaorg/commit/9b227c5921f4fd55a9dc1529a7fc5c128d3da83a

- http://schema.org/credentialOffered
- http://schema.org/Credential
- http://schema.org/CredentialInstance
- http://schema.org/credential
- http://schema.org/credentialIssuer
- http://schema.org/credentialIssuee
- http://schema.org/CredentialType
- http://schema.org/credentialType
- http://schema.org/Apprenticeship
- http://schema.org/ApprenticeshipType
- http://schema.org/_Apprenticeship
- http://schema.org/Certificate
- http://schema.org/CertificateType
- http://schema.org/JobSkillsCertificate
- http://schema.org/IndustryJobCertificate
- http://schema.org/Certification
- http://schema.org/CertificationType
- http://schema.org/AcademicDegree
- http://schema.org/AcademicDegreeType
- http://schema.org/SecondarySchoolDegree
- http://schema.org/AssociateDegree
- http://schema.org/BachelorDegree
- http://schema.org/GraduateDegree
- http://schema.org/DoctorateDegree
- http://schema.org/NanoDegree
- http://schema.org/Diploma
- http://schema.org/MilitaryCredential
- http://schema.org/MicroCredential
- http://schema.org/DigitalBadgeCredential
- http://schema.org/License
- http://schema.org/LicenseType
- http://schema.org/OccupationalLicense
- http://schema.org/LawLicense
- http://schema.org/MedicalLicense
- http://schema.org/TeachingLicense

.

- The Enumerations are derived from these discussions and mostly from https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl (CC-By-SA 4.0)
  - http://schema.org/credentialType <CredentialType>
  - http://schema.org/CredentialType
  - http://schema.org/ApprenticeshipType
  - http://schema.org/CertificateType
  - http://schema.org/CertificationType
  - http://schema.org/AcademicDegreeType
  - http://schema.org/LicenseType

- I changed ``grantsCredential`` to ``credentialOffered`` (d: {Course, +CourseInstance})
  - This is consistent with a ``credential-`` prefix for credential-related properties:
    - http://schema.org/credentialOffered
    - http://schema.org/credential
    - http://schema.org/credentialIssuer
    - http://schema.org/credentialIssuee
    - http://schema.org/credentialType

- This shell command finds classes added in a diff:
```bash
git diff HEAD^1 | grep '^\+' | grep 'resource=' | pyline -r '(.*)resource="(.*)"' 'rgx and ("- %s" % rgx.groups()[1])' | pbcopy
```

**Comment by westurner** [comment]
These snippets made it much easier to define these schema.org classes and properties; they're probably reusable:
- https://github.com/westurner/dotvim/blob/master/snippets-ulti/html.snippets

**Comment by westurner** [comment]
(I did ``git commit --amend; git push -f``'ed in order to keep a clean patch (after adding {Credential, CredentialInstance} to the domainIncludes of the valid[_] properties:

- http://schema.org/validFor
- http://schema.org/validFrom
- http://schema.org/validIn
- http://schema.org/validThrough
- http://schema.org/validUntil

I've updated the link to the commit in this https://github.com/schemaorg/schemaorg/issues/195#issuecomment-227852385 (which has probably already been emailed out.):

- https://github.com/westurner/schemaorg/commit/1f268fc1d1e85b7ebd205d136745289a919c9a34

**Comment by philbarker** [comment]
@westurner I think trying to define the different types of credential is very problematic. To take just one example:  If you look at https://en.wikipedia.org/wiki/Diploma you will see the range of different things that is covered by just one of these terms.

**Comment by pmika** [comment]
This is an automatically generated message.

pmika@yahoo-inc.com is no longer with Yahoo! Inc.

Your message will not be forwarded.

If you have a sales inquiry, please email yahoosales@yahoo-inc.com and someone will follow up with you shortly.

If you require assistance with a legal matter, please send a message to legal-notices@yahoo-inc.com

Thank you!

**Comment by westurner** [comment]
@philbarker re: Diploma:
- https://github.com/westurner/schemaorg/commit/1f268fc1d1e85b7ebd205d136745289a919c9a34#diff-d54dcb303dfa33fefe39415f7106623fR345
- A Diploma is a CredentialInstance (linking between a Person and an AcademicDegree)
  - CredentialInstance > Diploma
- [ ] Is a Diploma a Certificate?
  - Credential > Certificate > Diploma ?
- [ ] What additional properties beyond those defined as properties of Credential does/would a Diploma have?
- [ ] create a JSONLD example of Diploma
  - [x] create a JSOND example of Credential
    - [ ] push a commit full of examples (hopefully later this afternoon)

**Comment by westurner** [comment]
@CTI @CTITechPlanning : schemaorg / SKOS
- This'll need a more proper citation: https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl
  - [ ] The credType descriptions are much better than these
    - https://github.com/westurner/schemaorg/commit/1f268fc1d1e85b7ebd205d136745289a919c9a34#diff-d54dcb303dfa33fefe39415f7106623fR147
- With schema.org we don't have SKOS, but do have Enumerations
  - These Enumerations are also nested (and capitalized)
  - Two ways to merge efforts here:
    - a. CTI moves to schema.org/CredentialType Enumeration URLs for ``credType`` (here ``credentialType`` ("consistent with a credential- prefix for credential-related properties"))
    - b. Add ``sameAs``
      - e.g. ``credType:nanodegree`` ``sameAs`` ``schema:NanoDegree``

**Comment by danbri** [comment]
(Bookmarking a link)

I believe we need to sync the version in data/ext/pending/issue-195.rdfa with https://github.com/philbarker/schemaorg-1/commits/feature/ext-course (or the last few commits at least). /cc @westurner @philbarker

**Comment by westurner** [comment]
On Aug 1, 2016 8:08 AM, "Dan Brickley" <notifications@github.com> wrote:
>
> (Bookmarking a link)
>
> I believe we need to sync the version in data/ext/pending/issue-195.rdfa
with https://github.com/philbarker/schemaorg-1/commits/feature/ext-course
(or the last few commits at least). /cc @westurner @philbarker

Is there a way to leave the filename path for the schema (SCH) and example
(DAT) the same so that the rev history isnt broken?

**Comment by westurner** [comment]
re: subjectOfStudy
should it be
educationalAlignmentPrimary?
- educationalAlignment r: EducationalAlignment
- educationalAlignmentPrimary

Should it be r: [URL] or r: [URL, Text] (or r: EducationalAlignment) ?

**Comment by philbarker** [comment]
@westurner there's a summary of how we propose to deal with describing the subject of a course at https://www.w3.org/community/schema-course-extend/wiki/Subject_of_Course along with a link to the discussion. The consensus was that existing properties in CreativeWork had it covered.

On the question file names and revision history:  I haven't changed any filenames. I don't know what the best approach to merging the final version in schema will be.

**Comment by westurner** [comment]
Thanks @philbarker!

> The about property should be used where a single word or short phrase is used, the description property should be used for an unstructured sentence or paragraph of subject information. educationalAlignment may be used to supplement this with more precise reference to educational outcomes.

- http://schema.org/about r: http://schema.org/Thing
  - Thing.name
  - Thing.url
- http://schema.org/description r: http://schema.org/Text

.

- http://schema.org/description for sentences and paragraphs
- http://schema.org/about for words and phrases (~tags, #tags)
- http://schema.org/educationalAlignment for "An alignment to an established educational framework."
  - http://schema.org/AlignmentObject
    - alignmentType r:Text
      {teaches / educationalSubject,} ?
    - targetUrl r:URL
    - targetName r:Text
    - targetDescription r:Text
    - educationalFramework r:Text
      - [ ] can i add a URL here¡? (could/should it be {Text, Thing, URL})? @LRMI
   

IIUC, If one wanted to categorize courses with e.g. Wikipedia/dbpedia, Wikidata URIs, they could:

```
Course
name: Schema.org Course Metadata I
description: "Course Description"
about:
   name: "Schema.org"
   url: http://schema.org
   url: https://en.wikipedia.org/wiki/Schema.org
about:
   name: "RDFa"
   url: https://en.wikipedia.org/wiki/RDFa

educationalAlignment
   alignmentType: "teaches"
   targetName: "Schema.org Course Metadata"
   targetUrl: http://schema.org/Course
   targetUrl: http://schema.org
   targetUrl: https://schema.org
   targetUrl: https://en.wikipedia.org/wiki/Schema.org
```

IIUC, one use case for ``educationalAlignment`` is searching for course materials which teach a subject

One use case for ``about`` is searching for Things which are about something
So, a search query might want to search {``about``, ``description``, and ``educationalAlignment``} (or just fulltext query the whole JSONLD record)?

With consistent URIs/URLs in educationalFramework, it would be easier to present a **faceted search** for educationalFramework-specific learning resources

We could create a wiki page for Enumerating these EducationalFramework(s) with at least:
- name
- url
- .
- EnumerationSlug: e.g. CommonCore

- [ ] Is there a page with URIs for/of educationalFramework(s)? @LRMI
  - educationalFramework: "Common Core"
  - educationalFramework: "common core"
  - educationalFramework: "CommonCore"
  - educationalFramework: http://www.corestandards.org
  - educationalFramework: http://www.corestandards.org/ELA-Literacy/
  - educationalFramework: http://www.corestandards.org/Math/
  - educationalFramework: https://en.wikipedia.org/wiki/Common_Core_State_Standards_Initiative
  - educationalFramework: http://dbpedia.org/resource/Common_Core_State_Standards_Initiative
  - educationalFramework: http://dbpedia.org/page/Common_Core_State_Standards_Initiative
  - educationalFramework: http://dbpedia.org/resource/Common_Core ("http://dbpedia.org/ontology/wikiPageRedirects")


- [x] Is there a way to say this is an "#OER" (Open Educational Resource)? @LRMI
  - ``isAccessibleForFree`` r: {Event, CreativeWork}
      https://github.com/schemaorg/schemaorg/issues/900

**Comment by danbri** [comment]
Update on Courses status w.r.t. schema.org core. 

I asked @philbarker (as chair of https://www.w3.org/community/schema-course-extend/ ) yesterday for a list of the most basic Course-related terms that are approaching stability. It is important to show tangible progress and to stabilize the building blocks that we will need later.

We agreed that 

* Course + properties: courseCode, coursePrerequisites, hasCourseInstance
* CourseInstance + properties: courseMode, instructor

... are at a stage where it is useful to ask the Course schemas CG to double-check the definitions, examples and details before moving to final review by the wider schema.org community and its steering group.

In addition I've included /educationalCredentialAwarded which is currently more under discussion, since these definitions for now are all in the "pending" area. I'd love to move the leading 7 terms into schema.org core in our next release - does that sound feasible?

Regarding mechanics (which we could discuss in a separate Github issue if needed) I have copy/pasted manually from https://github.com/philbarker/schemaorg-1/tree/feature/ext-course/data/ext/course into https://github.com/schemaorg/schemaorg/blob/sdo-callisto/data/ext/pending/issue-195.rdfa for two reasons: 1.) it makes me read the definitions 2.) currently our implementation of the pending extension requires custom declarations per-term. This could be improved in the future, e.g. by moving the required declarations into a separate block of RDFa.

Are we ready to propose that the definitions for [Course](http://schema.org/Course), [courseCode](http://schema.org/courseCode), [coursePrerequisites](http://schema.org/coursePrerequisites), [hasCourseInstance](http://schema.org/hasCourseInstance), [courseMode](http://schema.org/courseMode), [instructor](http://schema.org/instructor) be moved into schema.org's next core release?

**Comment by danbri** [comment]
An implementation detail, of the kind I was hoping not to bore you all with:

I have moved the "pending" declarations of the Course terms out into a separate file. This means that https://github.com/schemaorg/schemaorg/blob/sdo-callisto/data/ext/pending/issue-195.rdfa  has less stuff in it that relates to the (hopefully transient) status of these terms w.r.t. pending.schema.org / pending.webschemas.org. Those details are kept in 
https://github.com/schemaorg/schemaorg/blob/sdo-callisto/data/ext/pending/boilerplate-195.rdfa instead.

**Comment by Gloyns** [comment]
Apologies for the noob question, however I'm looking to implement the course markup on a page where multiple courses are listed and, having scoured multiple sources, I can't find examples of this.

Google's example search experience here...

https://developers.google.com/search/docs/guides/search-gallery#courses

...shows a user searching by a broader course category, E.g. 'Android Courses'  with 3 courses listed from various providers.

The example code provided however only lists a single course.

Can anybody point me towards example markup for a category page where multiple courses are listed?

**Comment by danbri** [comment]
@Gloyns  - just repeat for each course. This following markup seems OK, although there might be some other JSON-LD idiom that allows the context declaration to appear just once at the top. Anyway:

    <script type="application/ld+json">
    [{
      "@context": "http://schema.org",
      "@type": "Course",
      "name": "Introduction to Computer Science and Programming",
      "description": "Introductory CS course laying out the basics.",
      "provider": {
        "@type": "Organization",
        "name": "University of Technology - Eureka",
        "sameAs": "http://www.ut-eureka.edu"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "Course",
      "name": "Postconstructivist desituationism and neomaterialist discourse",
      "description": "In the works of Joyce, a predominant concept is the distinction between
         masculine and feminine. Thus, if the conceptual paradigm of reality holds, we
         have to choose between neomaterialist discourse and neodialectic materialism.",
      "provider": {
        "@type": "Organization",
        "name": "DEPARTMENT OF LITERATURE, CARNEGIE-MELLON UNIVERSITY",
        "sameAs": "http://example.org/cmu/"
      }
    }]
    </script>

**Comment by Gloyns** [comment]
@danbri Many thanks, apologies - simply my lack of understanding of the correct syntax :) Thank you

**Comment by karindpowell** [comment]
@Gloyns 


MULTIPLE COURSE EXAMPLE

<script type="application/ld+json">
{
 "@context": "http://schema.org",
 "@type": "ItemList",
 "itemListElement" : [
 {
 "@type": "ListItem",
 "position": 1,
 "url" : "http://www2.sdhc.k12.fl.us/LYNX/HiTecIFrameKDP.asp?CourseID=208&ProgramID=10",
 "item": 
{
 "@type": "Course",
 "name": "H170690 NURSING ASSISTANT (CNA)",
"courseCode" : "H170690",
"image" : "http://www2.sdhc.k12.fl.us/assets/img/logo_lamp.png",
"grantsCredential" : "Certified Nursing Assistant (CNA)",
"previewUrl" :  "http://www2.sdhc.k12.fl.us/LYNX/HiTecIFrameKDP.asp?CourseID=208&ProgramID=10",
 "description": "This program prepares students for employment in a variety of health care settings including doctor’s offices, walk-in clinics, outpatient facilities, hospitals, and urgent care centers. The content includes, but is not limited to, interpersonal skills, medical terminology, legal and ethical responsibilities, safe and efficient work, gerontology, nutrition, pet-facilitated therapy, health and safety including Cardio-pulmonary Resuscitation (CPR) – heart saver level, and employability skills.",
 "provider": {
   "@type": "Organization",
   "name": "School District of Hillsborough County Florida - Technical Programs",
   "sameAs": "http://www.sdhc.k12.fl.us/doc/list/hillsborough-technical-colleges/program-areas/103-823/"
 }
}
},
{
"@type": "ListItem",
"position": 2,
"url" : "http://www2.sdhc.k12.fl.us/LYNX/HiTecIFrameKDP.asp?CourseID=243&ProgramID=10",
"item":
{
 "@context": "http://schema.org",
 "@type": "Course",
"courseCode" : "H170605",
"image" : "http://www2.sdhc.k12.fl.us/assets/img/logo_lamp.png",
"grantsCredential" : ["Certified Nursing Assistant (CNA)", "Licensed Practical Nurse (LPN)"],
"previewUrl" : "http://www2.sdhc.k12.fl.us/LYNX/HiTecIFrameKDP.asp?CourseID=243&ProgramID=10",
 "name":"H170605 PRACTICAL NURSING (LPN)",
 "description": "The program is designed to enable individuals to achieve instructional objectives in the following content areas: nursing skills, life span, nutrition, anatomy and physiology, medical nursing, surgical nursing, pediatric nursing, geriatric nursing, personal and community health, pharmacology, employability skills, leadership and human relations skills. Fifty percent of the training is spent in the clinical areas and includes: maternal and newborn, pediatrics, medical/surgical, and experience in extended care facilities. Clinical placement is determined through the Coordinating Council, may begin as early as 6:30 am and include out of county facilities. Successful program completion prepares students for the National Council Licensure Examination for Florida (NCLEX). Students completing the program receive credits toward a Registered Nurse A.S. Degree at Hillsborough Community College or any community college in the statewide system. Please contact the colleges for additional information.",
 "provider": {
   "@type": "Organization",
   "name": "School District of Hillsborough County Florida - Technical Programs",
   "sameAs": "http://www.sdhc.k12.fl.us/doc/list/hillsborough-technical-colleges/program-areas/103-823/"
 }
}
}
]
} 
</script>

**Comment by kaishiro** [comment]
Can I ask where the group netted out on the certification definition conversation that occurred upstream a bit? Couldn't find updates on any related issues but don't want to clog the queue if there is an active conversation ongoing. Lack of a certification schema seems like it should warrant an active effort.

**Comment by westurner** [comment]
- Ctrl-F for "credential" in this issue thread

- grantsCredential/credentialOffered implied a need for schema:Credential

-
https://github.com/westurner/schemaorg/commit/1f268fc1d1e85b7ebd205d136745289a919c9a34#diff-d54dcb303dfa33fefe39415f7106623fR147

- This is CTI schemaorg-ified
  - We'd need to revisit the latest CTI specs
- This patch is on file that's been moved
- It was suggested that this patch is insufficient and there
could/should/must be an additional Credential committee community group
list wiki and git feature branch (and so at least one GitHub issue with a
description to keep updated) as is the W3C way:
  - C: CreativeWork > Credential
  - C: CreativeWork > [Credential >?] CredentialInstance
    - There are possible Credentials, and there are instances of
credentials held by Organizations and Persons
    - P: CredentialInstance.credential r: Credential

**Comment by thadguidry** [comment]
@danbri @philbarker

I created the "online course" type on Wikidata https://www.wikidata.org/wiki/Q27885123 (and a subclass of course https://www.wikidata.org/wiki/Q600134 ) so far has 1 equivalent class linked to http://edamontology.org/data_3670

But reading through all the above and seeing what is in Pending...

1. Will the "online" indicator of a course be equitable through the courseMode property https://pending.schema.org/courseMode with a value of "online" ?

2. Or will "online" be designated by using the learningResourceType property https://pending.schema.org/learningResourceType with a value of "online" ?

3. or can we just have additionally a simple subtype for "OnlineCourse" to make all the publishers lives easier still ?

4. or a property on Course for the holding the idea of "remote learning" or "online" somehow ?

The reason I ask is that many folks will want to skip using courseMode and just have a quick filter for "show me only online courses for Calculus" , "remote learning for Philosophy".  These were a few sample queries I saw against Google Search.

**Comment by karindpowell** [comment]
@thadguidry I've been dwelling on this and here's my thoughts -- throwing them out there for comments...

In reference to your question regarding "courseMode": It looks to me like courseMode is more for "where your body is located when you take the course". In that case: "online"; "classroom", "lab", "hybrid" (online but sometimes in classroom); "field" (outside of classroom). These seem like some of the possibilities...

In reference to your question regarding "learningResourceType": It looks to me like learningResourceType is more for "what sort of **medium** is used by the instructor to get the knowledge into you". In that case: "slideshow"; "video"; "handouts"; "mixed media"; "book";

That said, let's say you had an online course. You might want the courseMode to be "online" and the learningResourceType might be "video" AND "book" AND "downloadable content".

**Comment by philbarker** [comment]
@thadguidry as it is possible for a course to be delivered online/through distance learning and face to face (i.e. as two CourseInstances of the same Course), I think that 1 works anf that 2 and 4 don't (which I think is what @karindpowell is getting at). For Courses I think that learningResourceType would best be used to distinguish between short courses, modules, programs etc, but that hasn't been widely discussed.

**Comment by westurner** [comment]
I think we need Enumerations and/or a wiki page with URIs for common values
for courseMode and learningResourceType. (that'll make it much easier to
build e.g. faceted interfaces with having to lowercase and spell-check)

**Comment by vholland** [comment]
To @philbarker's question, yes, I've seen cases where the same CourseInstance was both in person and online for distance learners. This is part of why we avoided an OnlineCourse type; it is really hard to define what is and is not "online".

**Comment by inetbiz** [comment]
I'd definitely like to see CourseSection. The LMS software I'm hacking had the ability like most lms software to preview free course units. And, if the consumer actually shares that free course unit, I'd like it defined properly as a CourseSection unit related to its parent Course.

**Comment by philbarker** [comment]
@inetbiz in what way is CourseSection distinct from CourseInstance ( https://pending.schema.org/CourseInstance An instance of a Course offered at a specific time and place or through specific media or mode of study or to a specific section of students. )

**Comment by inetbiz** [comment]
@philbarker I akin CourseSection to an article in a section. The course section is content that could stand on it's own outside of the course. Right now, I'm creating course content for HTML5 Semantic Structures. The course title is "Semantic SEO For Website Structures Course" which also includes RDFa, Microdata vocabularies.

**Comment by philbarker** [comment]
@inetbiz OK, I think I understand. There scope for confusion because CourseSection can be defined as "	A setting in which organized instruction of course content is provided to one or more students for a given period of time. " -- https://ceds.ed.gov/dataModelEntities.aspx 

I think you mean that you want to say that some material (or a sequence of one or more educational events and/or creative works) is a part of a Course?

As it stands you can say that Courses are part of Courses. I agree that how we model the distinction between a Programme of Study, a Module, a Course Unit, a Track etc needs more attention, but they all fit the definition of Course. Currently it is possible to use the learningResourceType property to say which of these you would call your course, but you'll find definitions & understanding of terms like 'module' or 'programme' vary greatly.

**Comment by inetbiz** [comment]
For now, I will use article and section. It just would be nice to give distinction within the vocabulary of Course. Your first paragraph https://github.com/schemaorg/schemaorg/issues/195#issuecomment-276707102 seems more of an instance.

**Comment by fravarese** [comment]
Hello everybody, I'm currently experimenting schema "courses" on my website, but I'm not sure about how should the google snippet look like .. In the official example page, it should look like a list:  [https://developers.google.com/search/docs/guides/search-gallery#courses ](url) ,  but in my pages this kind of list does not appear, so I wonder if I'm missing something.. even if  actually that kind of list does not appear neither in the website used as example (coursera.com), when appear in the google results page, among other search results. So I ask you, should that kind of listing appear in the google snippet once you implement schema in the code of a website or the example is worng and the list is just not supposed to apear? thank you very much. Francesco

**Comment by philbarker** [comment]
@fravarese I'm not sure this is the best place to discuss individual cases, but if you send me a link to your pages I'll have a look. phil.barker [at] hw.ac.uk

However I suspect that how your pages look in Google search engine results depends more on Google than the schema.org Course spec, and that I cannot help with.

**Comment by fravarese** [comment]
@philbarker 
Ok thanks, I tought there was a specific connection beetwen google and schema. I saw that in schema "events" the list-kind snippet always appears, and I tought that in "courses" should appear too.. because the example page showed it.

**Comment by westurner** [comment]
Release notes:
- http://blog.schema.org/2017/03/schemaorg-32-release-courses-fact.html
- https://schema.org/docs/releases.html#v3.2

Outstanding:

- [x] Event > CourseInstance
- [ ] [CreativeWork](https://schema.org/CreativeWork) > CourseInstance
  - Is there a reason this doesn't work?
  - Each CourseInstance could/*should* have properties from CreativeWork, as well? e.g.:
    - https://schema.org/accessModeSufficient
    - https://schema.org/accessibilityFeature
    - https://schema.org/accessibilitySummary
    - https://schema.org/contributor
    - https://schema.org/license
    - https://schema.org/isAccessibleForFree
    - https://schema.org/sourceOrganization

**Comment by RichardWallis** [comment]
> CreativeWork > CourseInstance
> Is there a reason this doesn't work?
> Each CourseInstance could/should have properties from CreativeWork, as well? 


As submitted in the proposal and defined in the pending area of the previous release (3.1) of Schema.org, _CourseInstance_ is not a subtype of _CreativeWork_ - its only super-type being _Event_.

I seem to remember that the logic behind this was discussed in detail within the [Course Extend Group ](https://www.w3.org/community/schema-course-extend/)well before the proposal was submitted.  @philbarker could probably clarify that for you.

**Comment by danbri** [comment]
Ah yep, this was our way around capturing the slightly ambiguous nature of "course". There's a type-like notion of instantiation. If specific properties are needed they can be attached directly and we tweak their definitions accordingly.

**Comment by westurner** [comment]
>  If specific properties are needed they can be attached directly and we tweak their definitions accordingly.

- these: https://github.com/schemaorg/schemaorg/issues/195#issuecomment-288920391
- https://schema.org/accessibilityHazard
  - "Accessibility of places" https://github.com/schemaorg/schemaorg/issues/254
    - "LocationFeatureSpecification"
      - https://schema.org/LocationFeatureSpecification

.

- What happened with courseUnits/courseCreditsUnit?
  - https://schema.org/PropertyValue > courseUnits (d: {Course, CourseInstance}) ?
    - unitCode: https://example.org/Col
      - NoCredit, Quarter, Semester, CarnegieUnits, ContinuingEducationUnits, ClockHours, Other
        - https://docs.google.com/document/d/12YWjLzZC8FiTiOwSAETRIEozeqZdn6O8a4fgqK4t5Ss/edit

- CourseDigitalFormats enumerations
  - Was "search for {SCORM,}" CourseInstances removed from the use cases?
    - https://www.w3.org/community/schema-course-extend/wiki/Outline_use_cases
      - How do we test that these assertions are now true?

**Comment by philbarker** [comment]
@westurner , all:
- The accessibility properties of a CourseInstance were not implemented because accessibility of all types of Event should be addressed by people who understand accessibility, not by people who understand educational courses.
- We didn't fully discuss credits for Courses. The current suggestion of attaching courseCredit and courseCreditUnits directly to Course does not work if you offer several options of credit. We need something a little bit similar to http://pending.webschemas.org/CategoryCode to make sure that the value and the units are associated with each other instead of having a choice of n values and n units and not knowing which goes with which. MLO Credit might be a model to follow ftp://ftp.cen.eu/CEN/Sectors/TCandWorkshops/Workshops/CWA16077.pdf but we need to check that such a model fits with current practice in the US (e.g. CEDS) and work of organizations like the Credential Transparency Initiative.  I think this is relatively easy and could be a sensible next step for the schema course extension community group.
- AFAIK "search for {SCORM}" never was on that list of use cases.
- Testing those assertions would need a body of resources that had been marked up. Guidance on how to provide data that would allow those tests would be a useful thing to create.

Hope that clarifies.

**Comment by westurner** [comment]
@philbarker Thanks!

> The accessibility properties of a CourseInstance were not implemented because accessibility of all types of Event should be addressed by people who understand accessibility, not by people who understand educational courses.

https://github.com/schemaorg/schemaorg/issues/254 "Accessibility of Places" (and Events (e.g. 'this event will have a language_xyz interpreter; but in general the Place is empty'))

> We didn't fully discuss credits for Courses. The current suggestion of attaching courseCredit and courseCreditUnits directly to Course does not work if you offer several options of credit. We need something a little bit similar to http://pending.webschemas.org/CategoryCode to make sure that the value and the units are associated with each other instead of having a choice of n values and n units and not knowing which goes with which.

If a URL is specified for CategoryCode.inCodeSet, what contentType is expected to be resolvable at that URL?
If a CategoryCodeSet is specified, CategoryCodeSet.hasCategoryCode is set.

> MLO Credit might be a model to follow ftp://ftp.cen.eu/CEN/Sectors/TCandWorkshops/Workshops/CWA16077.pdf but we need to check that such a model fits with current practice in the US (e.g. CEDS) and work of organizations like the Credential Transparency Initiative.

### CEDS

- Terms
  - https://ceds.ed.gov/Glossary.aspx
  - https://ceds.ed.gov/dataModelEntities.aspx
  - "CourseSection" ~= https://schema.org/CourseInstance
- "CourseCreditUnits"
  - K12: https://ceds.ed.gov/CEDSElementDetails.aspx?TermxTopicId=28729
  - Pst: https://ceds.ed.gov/CEDSElementDetails.aspx?TermxTopicId=28161
  - Values
    - NoCredit
    - Quarter
    - Semester
    - Units
    - CarnegieUnits
    - ContinuingEducationUnits
    - ClockHours
    - Other
    - Unreported
- "CourseCreditValue" (CarnegieUnits)
  - K12: https://ceds.ed.gov/CEDSElementDetails.aspx?TermxTopicId=31156
  - Pst: ...
- "NumberOfCreditsAttemped"
  - K12: ...
  - Pst: ...
- "NumberOfCreditsEarned"
  - K12: https://ceds.ed.gov/CEDSElementDetails.aspx?TermxTopicId=28513
  - Pst: ...
- https://schema.org/docs/schema_org_rdfa.html

### CTI
@CTITechPlanning 
- see next comment


> I think this is relatively easy and could be a sensible next step for the schema course extension community group.

- [ ] courseCredits (CategoryCode, Offer, OfferCatalog)
- [ ] Event > CourseInstance accessibility\*
- [ ] syllabus as a tree of Things / CreativeWorks (CourseUnit > CourseInstance, "ThingSequence", )
- [ ] Credential / CredentialInstance
  - [ ] W3C Community Group?
    - see next comment

**Comment by westurner** [comment]
From https://github.com/blockchain-certificates/cert-schema/issues/25#issuecomment-289006903 :

"Credential" as the super type:
- CTI: https://github.com/CredentialTransparencyInitiative/vocabularies/blob/master/credentialType.ttl
- https://github.com/schemaorg/schemaorg/issues/195#issuecomment-227852385
  - https://github.com/westurner/schemaorg/commit/1f268fc1d1e85b7ebd205d136745289a919c9a34 
    - Credential (that could be earned)
    - CredentialInstance (issuer={Person/Organization}, issuee={P/O}, validFrom, validUntil, ...)
    - e.g. Credential > Certificate > Diploma
      e.g. Credential > AcademicDegree > BachelorDegree
      e.g. Credential > Certification
      - note that https://schema.org/Course was released today (without grantsCredential or Credential yet)
        - https://schema.org/docs/releases.html#v3.2
          - http://blog.schema.org/2017/03/schemaorg-32-release-courses-fact.html

WebID
- | Wikipedia: https://en.wikipedia.org/wiki/WebID
- | Spec: https://www.w3.org/2005/Incubator/webid/spec/
- | Spec: https://www.w3.org/2005/Incubator/webid/spec/identity/
  - https://www.w3.org/2005/Incubator/webid/spec/identity/#publishing-a-webid-profile-using-the-rdfa-html-notation
- The Cert Ontology 1.0

The Cert Ontology 1.0 :
- https://www.w3.org/ns/auth/cert#
  - https://www.w3.org/ns/auth/cert#Certificate
    - foaf:Document > cert:Certificate > {cert:X509Certificate, cert:PGPCertificate}
    - "A certificate is a Document that is signed"
  - http://lov.okfn.org/dataset/lov/vocabs/cert

**Comment by philbarker** [comment]
@westurner thanks for information & comments, I'll try to work out what actions to take :)

**Comment by westurner** [comment]
> Testing those assertions would need a body of resources that had been marked up. Guidance on how to provide data that would allow those tests would be a useful thing to create.

- https://github.com/schemaorg/schemaorg/issues/53#issuecomment-289050849 "Linter warnings on schema.org examples"
  - https://github.com/schemaorg/schemaorg/pull/1491 "Travis linter"
    - https://github.com/schemaorg/schemaorg/blob/sdo-callisto/.travis.yml
- https://schema.org/Course#examples
  - To which file does one add an example (with e.g. "all possible attrs") now?
    - [ ] DAT: example of ``Course.hasCourseInstance`` with JSON-LD ``@reverse`` 
  - A Jupyter notebook with python.collections.OrderedDict, RDFa jinja2 and JSON-LD ``<script type="application/ld+json" id="jsonld">...</script>`` examples could be helpful
- https://www.w3.org/community/schema-course-extend/wiki/Outline_use_cases
  For the use cases which require search ("find", "refine"):
  - Is there a testable reference platform?
    - SPARQL queries of rdflib.Graph.parse(examples).query("WHERE")?
  - How / can I search for Courses with course attributes today?
    - Which search engines have implemented these schema.org Classes and Properties?

**Comment by jhoobergs** [comment]
How would I currently model an education programme ? I don't think that Course is right for this, so maybe a an EducationalProgramme definition ?

**Comment by philbarker** [comment]
@jhoobergs could you describe what you mean by an educational programme, preferably with links to a couple that you would like to model? Asking because the term is used in different ways.

**Comment by schivmeister** [comment]
1. Is there a recommendation as to when to use **CourseInstance** vs. **EducationEvent**? Some times this choice is not clear.

2. Also, is there a reason for there not being a relationship from **Course** to **EducationEvent**, like there is `hasCourseInstance`? Or is it expected that the link will be via `subEvent`? But that does not look right.

3. How are search engines presently interpreting these types and relationships? What about courses with related course instances or education events? It seems there would be a disconnect if you showed just courses without showing the events associated with it. Do SEs presently parse relationships or are they ignored? (e.g. Provider -> Course -> CourseInstance)

I think sooner or later we're going to have to deal with the diversity of educational programmes. What we typically refer to as "course" can be understood to be disjoint with what we consider as a "training programme". A course typically culminates in a "credential" but a training programme, while offering certificates "of participation" or similar, would not typically culminate in a credential. This is the case for IT training, where service providers typical offer training for certification exams. I think the same holds for professional certifications, where one would pursue some kind of training before taking the relevant exam(s).

**Comment by philbarker** [comment]
@schivmeister hello.
1. We haven't discussed any recommendation for choosing between EducationEvent and CourseInstance.
2. EducationEvent as a subEvent of a CourseInstance would make sense if you wanted to itemise the individual sessions (lectures, lab classes, lessons etc) that made up a CourseInstance.
3. I don't know.

On your general point, the intent was that Course would encompass as many scenarios as possible, and is defined so as not to be disjoint 'training programme'. The problem with starting with "what we typically refer to as "course", is that typical doesn't travel very well (indeed, even the narrow sector that I work in, UK Higher Education, it prove impossible to narrow down the term "course" in any sensible way based on how it is used, see https://www.hesa.ac.uk/files/What-is-a-course_2011.pdf ) Hence the definition "an educational course is a sequence of one or more educational events and/or creative works which aims to build knowledge, competence or ability of learners." 

I think you are right that there could be more done to describe the type of credentialing / certification that results from successfully taking a Course.  That's a can of worms in itself. It's also one that goes beyond describing Courses per se, because all sorts of accreditation bodies, professional bodies, employers, regulators and so on have a view on credentialing and certification.

**Comment by schivmeister** [comment]
@philbarker Thanks for responding to my points, you are absolutely right. My general point was in reference to https://github.com/schemaorg/schemaorg/issues/195#issuecomment-290936168 for modelling an education "programme". As you said, it is quite open-ended and the best course of action is to use what is available. I see now you discussed a similar issue before in https://github.com/schemaorg/schemaorg/issues/195#issuecomment-262537643.

My concern was really about the linkages between courses and events, since these two types are presently known to be parsed by search engines. I was a bit surprised that there is an absent link between **Course** and **EducationEvent**. I would have expected something like `hasRelatedEvent` on the former and/or `hasRelatedCourse` on the latter.

I was even more surprised that subclasses are defined without properties that are unique to them. From an ontology/vocabulary design perspective, http://schema.org/EducationEvent looks redundant to me, whereas http://schema.org/CourseInstance has its own properties that fully qualifies it as a subclass. But I suppose that's how things work in the specs, and perhaps just the type name is sufficient to help parsers interpret them differently.

**Comment by jhoobergs** [comment]
@philbarker By educational programma I mean thinks like you can see at https://onderwijsaanbod.kuleuven.be/opleidingen/e/CQ_50074163.htm#activetab=diploma_omschrijving or https://onderwijsaanbod.kuleuven.be/opleidingen/e/CQ_51854602.htm#activetab=diploma_omschrijving

**Comment by philbarker** [comment]
@vholland @danbri I think this issue can be closed. We've addressed most of the points raised and it would be easier to track the anything still missing through discrete issues.

**Comment by clouddevloper** [comment]
I have added the course schema. But my page is still not showing up in search reasults.

**Comment by westurner** [comment]
From DynoMapper  https://dynomapper.com/blog/265-using-rich-cards-for-search
:

> At this point in time, the content types that can have the rich card
features are: videos, recipes, movies, reviews, music, local businesses,
and articles. [...]

> Initially, rich cards were only available for movies and recipes as a new
way for website owners to give previews of their content within a search
results page. Recently, two more 'verticals' have been added for websites
based in the United States: online courses and restaurants.
